<launch>
    <arg name="node_start_delay" default="120.0"/>  
    <node pkg="rasa_ros" name="rasa_action" type="rasa_action.sh" output="screen"/>
    <node pkg="rasa_ros" name="rasa_server" type="rasa_server.sh" output="screen"/> 
    <node pkg="rasa_ros" name="dialogue_server" type="dialogue_server.py" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>

    <node pkg="project_work" type="pepper_start.py" name="PepperNode" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
    <node pkg="project_work" type="speech2text.py" name="speech2text_service" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
    <node pkg="project_work" type="audio_detected_node.py" name="audio_node" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
    <node pkg="project_work" type="audio_analysis_node.py" name="audio_analysis" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
    <node pkg="project_work" type="image_publisher.py" name="image_publisher" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
    <node pkg="project_work" type="face_detector_node.py" name="face_detector" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
    <node pkg="project_work" type="dialogue_node.py" name="dialogue_node" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
    <node pkg="project_work" type="engagement_node.py" name="engagement_node" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
    <node pkg="project_work" type="t2s_node.py" name="Text2Speech" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
    <node pkg="project_work" type="following_node.py" name="FollowingNode" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
    
</launch>
